encoding.bs

```diff
@@ -1305,6 +1350,404 @@ must run these steps:
 </ol>
 
 
+<h3 id=interface-mixin-generictransformstream>Interface Mixin {{GenericTransformStream}}</h3>
+
+<p>{{GenericTransformStream}} represents the concept of a <a>transform stream</a> in IDL. It is not
+a {{TransformStream}}, though it has the same interface and it delegates to one.
+
+<pre class=idl>
+interface mixin GenericTransformStream {
+  readonly attribute ReadableStream readable;
+  readonly attribute WritableStream writable;
+};
+</pre>
+
+<p>An object that includes GenericTransformStream always has an associated
+<dfn for=GenericTransformStream>transform</dfn>.
+
+<p>The <dfn attribute for=GenericTransformStream><code>readable</code></dfn> attribute's getter must
+return <a for=GenericTransformStream>transform</a>.\[[readable]].
+
+<p>The <dfn attribute for=GenericTransformStream><code>writable</code></dfn> attribute's getter must
+return <a for=GenericTransformStream>transform</a>.\[[writable]].
+
+
+<h3 id=interface-textdecoderstream>Interface {{TextDecoderStream}}</h3>
+
+<pre class=idl>
+[Constructor(optional DOMString label = "utf-8", optional TextDecoderOptions options),
+ Exposed=(Window,Worker)]
+interface TextDecoderStream {
+  USVString decode(optional BufferSource input, optional TextDecodeOptions options);
+};
+
+TextDecoderStream includes TextDecoderAttributes;
+TextDecoderStream includes GenericTransformStream;
+</pre>
+
+<p>A {{TextDecoderStream}} object has an associated <dfn for=TextDecoderStream>encoding</dfn>,
+<dfn for=TextDecoderStream>decoder</dfn>, <dfn for=TextDecoderStream>stream</dfn>,
+<dfn for=TextDecoderStream>ignore BOM flag</dfn> (initially unset),
+<dfn for=TextDecoderStream>BOM seen flag</dfn> (initially unset),
+<dfn for=TextDecoderStream>error mode</dfn> (initially "<code>replacement</code>"),
+<dfn for=TextDecoderStream>do not flush flag</dfn> (initially unset), and
+<dfn for=TextDecoderStream>transform</dfn>.
```

Comment:
For those things provided by the mixins, you should use the mixin `dfn`s, and for=TextDecoderAttributes on references to them. No need to have separate dfns/internal slots for the TextDecoder and TextDecoderStream themselves.